### DISCUSSION TITLE
trouble with the 'empty_option' parameter

### DISCUSSION THREAD

**Discussion by SebastiendOrnano** [sqlpage/SQLPage#712] [Q&A]
I want to use the 'empty_option' parameter in a 'select' type field but I does not work.
So I copy paste the example 'select a fruit' from the documentation in my index.sql file but I does not work.
I tried with an older version of sqlpage.bin but I does not work.
I cleaned my sqlpage.json  but I does not work.
My templates folder is empty.

So I do not understand where the problem is coming from. (nb : I work on a sqlite database / desktop on linux mint 22)

Any help is welcome and thanks in advance !

**Comment by lovasoa** [comment]
Hello,
  Let's debug your issue ! 

Can you follow these steps and report at which point you do not see what you would expect, and what you see instead.

```sh
mkdir /tmp/sqlpage_712
cd /tmp/sqlpage_712
wget https://github.com/sqlpage/SQLPage/releases/download/v0.31.0/sqlpage-linux.tgz
tar xvf sqlpage-linux.tgz
cat <<EOF > index.sql 
select 
    'form'                        as component;
select 
    'Fruit'             as name,
    'select'            as type,
    'Select a fruit...' as empty_option,
    '[{"label": "Orange", "value": 0}, {"label": "Apple", "value": 1}, {"label": "Banana", "value": 3}]' as options;
EOF
./sqlpage.bin
```

Then post a screenshot of what you see in your browser on http://localhost:8080, and what you see in your console.

This is what the console shows for me

```
❯ ./sqlpage.bin 
[2024-11-25T14:02:14.814Z INFO  sqlpage::webserver::database::connect] Connecting to database: sqlite://./sqlpage/sqlpage.db?mode=rwc
[2024-11-25T14:02:14.815Z INFO  sqlpage::webserver::database::migrations] No migration found in /tmp/xxx/sqlpage/migrations. You can specify database operations to apply when the server first starts by creating files in migrations/<VERSION>_<DESCRIPTION>.sql where <VERSION> is a number and <DESCRIPTION> is a short string.
[2024-11-25T14:02:14.829Z INFO  sqlpage::webserver::http] SQLPage v0.31.0 started successfully.
        Now listening on 0.0.0.0:8080: accessible from the network, and locally on http://localhost:8080
        You can write your website's code in .sql files in /tmp/xxx
```

![image](https://github.com/user-attachments/assets/f148d413-4b1b-4ce7-9b8e-aee54870556f)


If the results are different for you after following the exact same steps, then we have a bug. Otherwise, you have a problem in your configuration, probably similar to last time, when you had overwritten the prebuilt components without realizing it.

**Reply by SebastiendOrnano** [reply]
I have a parsing error but perhaps due to a copy paste error
![Capture d’écran du 2024-11-25 15-26-12](https://github.com/user-attachments/assets/69606ee4-4f4d-479d-ae3e-081fd86114ae)

This time, I checked to have a basic configuration (sqlpage.json, empty templates folder)

**Reply by SebastiendOrnano** [reply]
I retried but without the dot . before the first 'select' and I worked !

So the trouble is coming from my configuration. 
I will start again from scratch to eliminate any possible conflict.

Thank again for your quick answer !

**Reply by lovasoa** [reply]
Sorry about the dot, it was a typo!

**Reply by lovasoa** [reply]
Let me know when you find the source of the error, or if you need further help for debugging

**Reply by SebastiendOrnano** [reply]
I finally found something : as I am on linux mint as setup sqlpage as a service following your tuto and so my sqlpage.bin file was in the folder  /usr/local/bin
![Capture d’écran du 2024-11-26 09-10-06](https://github.com/user-attachments/assets/9d2de5c9-c1d7-4ece-9810-6725218e8024)
 and there  I had this problem with de empty_option parameter.

I rebuilt my database and the demo and I changed the location of the sqlpage.bin file to /var/www/sqlpage
![Capture d’écran du 2024-11-26 09-04-11](https://github.com/user-attachments/assets/815505a0-8be4-49a2-81fd-604b2b8ee5a0)
and there it works

It is above my competences to understand why it works there and not here. 
Before changing the location of the sqlpage.bin file, I asked my sql teacher (https://iask.ai/) who suggested a possible conflict with an environmental variable. In the www folder I found var reliated to snapd and flak, so I uninstalled completely snap and flackpack and from that point it seems to work properly in www. So I checked for possible environmental variables in /usr and there are still some variables relied to snap

![Capture d’écran du 2024-11-26 09-24-36](https://github.com/user-attachments/assets/334b3432-9b92-4c37-aac0-f9960be6d899)

But honestly it is a bit of chinese for me. The important thing is I found a way to make it works !

**Reply by lovasoa** [reply]
Are you sure that the problem wasn't just that you had updated the wrong `sqlpage.bin` file ? If you update the right file (the `/usr/local/bin/sqlpage.bin` file pointed to by ExecStart), and then restart the service with `sudo systemctl restart sqlpage`, then you should get the latest version running.
